File: frontend/src/services/userStory.ts
import api from './api'

export interface UserStory {
  story_id: string
  project_id: string
  title: string
  description: string | null
  acceptance_criteria: string | null
  page_desc: string | null
  priority: string
  status: string
  created_at: string | null
  updated_at: string | null
}

export interface UserStoryCreatePayload {
  title: string
  description?: string | null
  acceptance_criteria?: string | null
  page_desc?: string | null
  priority?: string
  status?: string
}

export interface UserStoryUpdatePayload {
  title?: string
  description?: string | null
  acceptance_criteria?: string | null
  page_desc?: string | null
  priority?: string
  status?: string
}

export async function listUserStories(projectId: string): Promise<UserStory[]> {
  const res = await api.get<UserStory[]>(`/v1/projects/${projectId}/user-stories`)
  return res.data
}

export async function createUserStory(projectId: string, payload: UserStoryCreatePayload): Promise<UserStory> {
  const res = await api.post<UserStory>(`/v1/projects/${projectId}/user-stories`, payload)
  return res.data
}

export async function getUserStory(storyId: string): Promise<UserStory> {
  const res = await api.get<UserStory>(`/v1/user-stories/${storyId}`)
  return res.data
}

export async function updateUserStory(storyId: string, payload: UserStoryUpdatePayload): Promise<UserStory> {
  const res = await api.patch<UserStory>(`/v1/user-stories/${storyId}`, payload)
  return res.data
}

export async function deleteUserStory(storyId: string): Promise<void> {
  await api.delete(`/v1/user-stories/${storyId}`)
}

export interface UserStoryImportResult {
  imported_count: number
  skipped_count: number
  stories: Array<{ story_id: string; title: string }>
}

export async function importUserStoriesFromRequirements(
  projectId: string,
): Promise<UserStoryImportResult> {
  const res = await api.post<UserStoryImportResult>(
    `/v1/projects/${projectId}/user-stories/import-from-requirements`,
  )
  return res.data
}

File: backend/app/schemas/user_story.py
"""Pydantic DTOs for UserStory operations."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class UserStoryCreateDTO(BaseModel):
    """Create user story request."""

    title: str = Field(..., min_length=1, max_length=256)
    description: str | None = Field(default=None)
    acceptance_criteria: str | None = Field(default=None)
    page_desc: str | None = Field(default=None, description="页面描述段落，用于草图生成")
    priority: str = Field(default="P1")
    status: str = Field(default="DRAFT")


class UserStoryUpdateDTO(BaseModel):
    """Update user story request."""

    title: str | None = Field(default=None, min_length=1, max_length=256)
    description: str | None = Field(default=None)
    acceptance_criteria: str | None = Field(default=None)
    page_desc: str | None = Field(default=None)
    priority: str | None = Field(default=None, pattern=r"^(P0|P1|P2|P3)$")
    status: str | None = Field(default=None, pattern=r"^(DRAFT|ACTIVE|ARCHIVED)$")


class UserStoryResponseDTO(BaseModel):
    """User story response."""

    story_id: str
    project_id: str
    title: str
    description: str | None
    acceptance_criteria: str | None
    page_desc: str | None
    priority: str
    status: str
    created_at: datetime | None
    updated_at: datetime | None


class UserStoryImportResultDTO(BaseModel):
    """Result of importing user stories from requirements."""

    imported_count: int
    skipped_count: int
    stories: list[dict[str, str]]
